internal/backends/sqlite/collection.go

```diff
@@ -110,30 +110,20 @@ func (c *collection) InsertAll(ctx context.Context, params *backends.InsertAllPa
 
 	db := c.r.DatabaseGetExisting(ctx, c.dbName)
 	meta := c.r.CollectionGet(ctx, c.dbName, c.name)
+	batchSize := 100
 
 	err := db.InTransaction(ctx, func(tx *fsql.Tx) error {
-		for _, doc := range params.Docs {
-			b, err := sjson.Marshal(doc)
-			if err != nil {
-				return lazyerrors.Error(err)
+		var j int
+		for i := 0; i < len(params.Docs); i += batchSize {
+			if j += batchSize; j > len(params.Docs) {
+				j = len(params.Docs)
 			}
```

Comment:
I think that could be simplifying by using a slice instead of a second index `j`:
```go
docs := params.Docs

// ...

i := min(batchSize, len(docs))
batch, docs := docs[:i], docs[i:]
```

Reply:
Thanks it's easier to read too